cpu.go

```diff
@@ -297,14 +299,43 @@ func (stat *CPUWatcher) Once() error {
 			default:
 				continue
 			}
-			values[fmt.Sprintf("%s.%%d.cpu%d", utype, i)] = value
-			values[fmt.Sprintf("%s.%%d.total", utype)] += value
+
+			if runtime.GOOS == "windows" {
+				// calculate the logical CPU index from "cpuIndex,coreIndex" format
+				cpuIndexParts := strings.Split(cputime.CPU, ",")
+
+				cpuIndex, _ := strconv.Atoi(cpuIndexParts[0])
+				coreIndex, _ := strconv.Atoi(cpuIndexParts[1])
+
+				if _, exists := cpuStatPerCpuPerCorePerType[cpuIndex]; !exists {
+					cpuStatPerCpuPerCorePerType[cpuIndex] = make(map[int]map[string]float64)
+				}
+
+				if _, exists := cpuStatPerCpuPerCorePerType[cpuIndex][coreIndex]; !exists {
+					cpuStatPerCpuPerCorePerType[cpuIndex][coreIndex] = make(map[string]float64)
+				}
+
+				// store by indexes to iterate in the right order later
+				cpuStatPerCpuPerCorePerType[cpuIndex][coreIndex][utype] = value
+				values[fmt.Sprintf("%s.%%d.total", utype)] += value / float64(len(times))
+
+			} else {
+				values[fmt.Sprintf("%s.%%d.%s", utype, cputime.CPU)] = value
+				values[fmt.Sprintf("%s.%%d.total", utype)] += value / float64(len(times))
+			}
 		}
 	}
 
-	for k, v := range values {
-		if strings.HasSuffix(k, ".total") {
-			values[k] = v / float64(len(times))
+	if runtime.GOOS == "windows" {
+		// calculate CPU logical index on Windows
```

Comment:
Could add a short description how this is done? Not in much details but give an idea about the following loops.

Reply:
Done!